src/main/java/org/jahia/modules/jcrestapi/JCRRestAPIDeprecationFilter.java

```diff
@@ -0,0 +1,81 @@
+package org.jahia.modules.jcrestapi;
+
+import org.jahia.osgi.BundleUtils;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A filter that logs deprecation warnings for deprecated REST API endpoints.
+ */
+@Provider
+@Priority(Priorities.USER)
+public class JCRRestAPIDeprecationFilter implements ContainerResponseFilter {
+
+    public static final Logger logger = LoggerFactory.getLogger(JCRRestAPIDeprecationFilter.class);
+    private static final Map<String, Long> loggedPaths = new ConcurrentHashMap<>();
+
+    @Context
+    private UriInfo uriInfo;
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
+        // Always send deprecation header:
+        responseContext.getHeaders().add("Deprecation", true);
+
+        // Load the config, aware that the implementation is a bit dirty and not OSGi compliant,
+        // but it's ok, remember that we are in deprecated code.
+        ConfigurationAdmin configAdmin = BundleUtils.getOsgiService(ConfigurationAdmin.class, null);
```

Comment:
Can we make it as a private reference, to avoid the service lookup at each api call?